tests/unittests/tests-ipv4_addr/tests-ipv4_addr.c

```diff
@@ -0,0 +1,164 @@
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+#include <errno.h>
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "byteorder.h"
+#include "net/ipv4/addr.h"
+
+#include "tests-ipv4_addr.h"
+
+static void test_ipv4_addr_equal__unequal(void)
+{
+    ipv4_addr_t a = { { 1, 1, 1, 0 } };
+    ipv4_addr_t b = { { 1, 1, 1, 1 } };
+
+    TEST_ASSERT(!ipv4_addr_equal(&a, &b));
+}
+
+static void test_ipv4_addr_equal__equal(void)
+{
+    ipv4_addr_t a = { { 1, 1, 1, 1 } };
+    ipv4_addr_t b = { { 1, 1, 1, 1 } };
+
+    TEST_ASSERT(ipv4_addr_equal(&a, &b));
+}
+
+static void test_ipv4_addr_to_str__string_too_short(void)
+{
+    ipv4_addr_t a = { { 1, 1, 1, 1 } };
+    char result[1];
+
+    TEST_ASSERT_NULL(ipv4_addr_to_str(result, &a, sizeof(result)));
+}
+
+static void test_ipv4_addr_to_str__addr_NULL(void)
+{
+    char result[IPV4_ADDR_MAX_STR_LEN];
+
+    TEST_ASSERT_NULL(ipv4_addr_to_str(result, NULL, sizeof(result)));
+}
+
+static void test_ipv4_addr_to_str__result_NULL(void)
+{
+    ipv4_addr_t a;
+
+    TEST_ASSERT_NULL(ipv4_addr_to_str(NULL, &a, IPV4_ADDR_MAX_STR_LEN));
+}
+
+static void test_ipv4_addr_to_str__success(void)
+{
+    ipv4_addr_t a = { { 1, 1, 1, 1 } };
+    char result[IPV4_ADDR_MAX_STR_LEN];
+
+    TEST_ASSERT_EQUAL_STRING("1.1.1.1", ipv4_addr_to_str(result, &a, sizeof(result)));
+}
+
+static void test_ipv4_addr_to_str__success2(void)
+{
+    ipv4_addr_t a = { { 0, 1, 12, 123 } };
+    char result[IPV4_ADDR_MAX_STR_LEN];
+
+    TEST_ASSERT_EQUAL_STRING("0.1.12.123", ipv4_addr_to_str(result, &a, sizeof(result)));
+}
+
+static void test_ipv4_addr_from_str__dot_start(void)
+{
+    ipv4_addr_t result;
+
+    TEST_ASSERT_NULL(ipv4_addr_from_str(&result, ".1.12.123"));
+}
+
+static void test_ipv4_addr_from_str__double_dot(void)
+{
+    ipv4_addr_t result;
+
+    TEST_ASSERT_NULL(ipv4_addr_from_str(&result, "0..12.123"));
+}
+
+static void test_ipv4_addr_from_str__string_too_long(void)
+{
+    ipv4_addr_t result;
+
+    TEST_ASSERT_NULL(ipv4_addr_from_str(&result, "255.255.255.255.255"));
+}
+
+static void test_ipv4_addr_from_str__illegal_chars(void)
+{
+    ipv4_addr_t result;
+
+    TEST_ASSERT_NULL(ipv4_addr_from_str(&result, "0.::ab-)"));
+}
+
+static void test_ipv4_addr_from_str__addr_NULL(void)
+{
+    ipv4_addr_t result;
+
+    TEST_ASSERT_NULL(ipv4_addr_from_str(&result, NULL));
+}
+
+static void test_ipv4_addr_from_str__result_NULL(void)
+{
+    TEST_ASSERT_NULL(ipv4_addr_from_str(NULL, "::"));
+}
+
+static void test_ipv4_addr_from_str__success(void)
+{
+    ipv4_addr_t a = { { 1, 1, 1, 1 } };
+    ipv4_addr_t result;
+
+    TEST_ASSERT_NOT_NULL(ipv4_addr_from_str(&result, "1.1.1.1"));
+    TEST_ASSERT(ipv4_addr_equal(&a, &result));
+}
+
+static void test_ipv4_addr_from_str__success2(void)
+{
+    ipv4_addr_t a = { { 0, 1, 12, 123 } }, result;
+
+    TEST_ASSERT_NOT_NULL(ipv4_addr_from_str(&result, "0.1.12.123"));
+    TEST_ASSERT(ipv4_addr_equal(&a, &result));
+}
+
```

Comment:
how about adding a test for conversion with the `inet_[pton|ntop]()`, like:
``` C
static void test_inet_pton_ntop__success(void)
{
    ipv4_addr_t result;
    size_t res_str_len = 11;
    char res_str[res_str_len];

    TEST_ASSERT_NOT_NULL(inet_pton(AF_INET, "0.1.12.123", &result));
    TEST_ASSERT_NOT_NULL(inet_ntop(AF_INET, &result, res_str, res_str_len));
}
```
It would provide certainty that the wrappers work as intended (which is the case) and extend the coverage of these unittests for future CI tests in case the code changes.